teaching_position.rb

```diff
@@ -0,0 +1,34 @@
+require_relative 'people'
+
+class TeachingPosition < People
+
+attr_accessor :num, :target_raise, :salary, :number
```

Comment:
This reads as "a Teaching position is a kind of person" which doesn't sound quite right. A good alternative would be to call the class name "Teacher" instead of "TeachingPosition"

```ruby
class ApprenticeTeacher < Teacher   -> Apprentice teacher is a Teacher
end

class SeniorTeacher < Teacher  -> Senior teacher is a Teacher
end

class Teacher < Person -> Teacher is a Person
end

class Student < Person   -> Student is a Person
end
```